pkg/streamingpromql/operators/binops/vector_scalar_binary_operation.go

```diff
@@ -94,26 +94,23 @@ func NewVectorScalarBinaryOperation(
 	return b, nil
 }
 
-func (v *VectorScalarBinaryOperation) SeriesMetadata(ctx context.Context) ([]types.SeriesMetadata, error) {
+func (v *VectorScalarBinaryOperation) SeriesMetadata(ctx context.Context) (types.SeriesMetadataSet, error) {
 	// Get the scalar values once, now, rather than having to do this later in NextSeries.
 	var err error
 	v.scalarData, err = v.Scalar.GetValues(ctx)
 	if err != nil {
-		return nil, err
+		return types.NewEmptySeriesMetadataSet(), err
 	}
 
 	metadata, err := v.Vector.SeriesMetadata(ctx)
 	if err != nil {
-		return nil, err
+		return types.NewEmptySeriesMetadataSet(), err
 	}
 
 	if !v.Op.IsComparisonOperator() || v.ReturnBool {
 		// We don't need to do deduplication and merging of series in this operator: we expect that this operator
 		// is wrapped in a DeduplicateAndMerge.
-		metadata, err = functions.DropSeriesName.Func(metadata, v.MemoryConsumptionTracker)
-		if err != nil {
-			return nil, err
-		}
+		return functions.DropSeriesName.Func(metadata, v.MemoryConsumptionTracker, false)
```

Comment:
Don't we need to pass `EnableDelayedNameRemoval` here? (rather than always passing false)